src/testrender/cuda/rend_lib.cu

```diff
@@ -20,56 +21,56 @@ OSL_NAMESPACE_EXIT
 extern "C" {
 
     __device__
-    void* closure_component_allot (void* pool, int id, size_t prim_size, const float3& w)
+    void* closure_component_allot (void* pool, int id, size_t prim_size, const OSL::Color3& w)
     {
-        ((ClosureComponent*) pool)->id = id;
-        ((ClosureComponent*) pool)->w  = w;
+        ((OSL::ClosureComponent*) pool)->id = id;
+        ((OSL::ClosureComponent*) pool)->w  = w;
 
-        size_t needed   = (sizeof(ClosureComponent) - sizeof(void*) + prim_size + 0x7) & ~0x7;
+        size_t needed   = (sizeof(OSL::ClosureComponent) - sizeof(void*) + prim_size + 0x7) & ~0x7;
```

Comment:
Could use `(alignof(OSL::ClosureComponent) - 1)` instead of `0x7` to be a bit more explicit?

Also, while I realize that all the users of the pool are probably the same alignment right now, this isn't necessarily true in general. I believe the CPU side aligns first and then figures out the size (see code in the `SimplePool`  class).

Reply:
Sounds good to me!

Aligning the pointer sounds interesting. I'll check out the SimplePool class and see what I can do.